src/plugins/intel_gpu/src/plugin/transformations/dynamic_quantize_fully_connected.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "dynamic_quantize_fully_connected.hpp"
+
+#include "intel_gpu/op/fully_connected_compressed.hpp"
+#include "intel_gpu/op/dynamic_quantize.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+#include "intel_gpu/runtime/debug_configuration.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+DynamicQuantizeFullyConnected::DynamicQuantizeFullyConnected(int64_t group_size) {
+    GPU_DEBUG_GET_INSTANCE(debug_config);
+    using namespace ov::pass::pattern;
+
+    // per-token quantization is supported
+    if (group_size != -1) {
+        GPU_DEBUG_TRACE << "Dynamic quantization is disabled " << group_size << std::endl;
+        return;
+    }
+
+    auto data = any_input();
+    auto fully_connected_compressed3 = wrap_type<op::FullyConnectedCompressed>({data, any_input(), any_input(), any_input()});
+    auto fully_connected_compressed4 = wrap_type<op::FullyConnectedCompressed>({data, any_input(), any_input(), any_input(), any_input()});
+    auto fully_connected_compressed = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{fully_connected_compressed3, fully_connected_compressed4});
+
+
+    ov::matcher_pass_callback callback = [=](Matcher& m) {
+        if (transformation_callback(m.get_match_root())) {
+            return false;
+        }
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto& m_data = pattern_map.at(data).get_node_shared_ptr();
+
+        std::shared_ptr<ov::intel_gpu::op::FullyConnectedCompressed> m_fc;
+        
+        if (pattern_map.find(fully_connected_compressed3) != pattern_map.end())
+            m_fc = std::dynamic_pointer_cast<op::FullyConnectedCompressed>(pattern_map.at(fully_connected_compressed3).get_node_shared_ptr());
+        else if (pattern_map.find(fully_connected_compressed4) != pattern_map.end())
+            m_fc = std::dynamic_pointer_cast<op::FullyConnectedCompressed>(pattern_map.at(fully_connected_compressed4).get_node_shared_ptr());
+
+        if (m_data->get_element_type() == ov::element::Type_t::f32) {
```

Comment:
Please move it to transformation callback

Reply:
do you mean to implement it as a lambda function from `transformations_pipeline.cpp`? Is it our policy to expose the criteria from `transformations_pipeline.cpp` instead of putting within the pass?

Reply:
That's not a strict requirement, but our direction is to reuse passes cross plugins as much as possible. So we need to move GPU specific limitation to external callback instead of hardcoding it inside the pass

Reply:
oh I see.

Reply:
applied, thanks!